examples/rvc-app/rvc-common/src/rvc-service-area-delegate.cpp

```diff
@@ -0,0 +1,415 @@
+    // We are reimplementing this method, still using linear search, but with some optimization on the SDK implementation
+    // since we have direct access to the list.
+    listIndex         = 0;
+    uint32_t listSize = static_cast<uint32_t>(mSupportedLocations.size());
+
+    while (listIndex < listSize)
+    {
+        if (mSupportedLocations[listIndex].locationID == aLocationId)
+        {
+            aSupportedLocation = mSupportedLocations[listIndex];
+            return true;
+        }
+
+        ++listIndex;
+    }
+
+    return false;
+};
+
+bool RvcServiceAreaDelegate::AddSupportedLocation(const LocationStructureWrapper & newLocation, uint32_t & listIndex)
+{
+    // The server instance (caller) is responsible for ensuring that there are no duplicate location IDs, list size not exceeded,
+    // etc.
+
+    // Double-check list size to ensure there no memory issues.
+    if (mSupportedLocations.size() < kMaxNumSupportedLocations)
+    {
+        // not sorting list, number of locations normally expected to be small, max 255
+        mSupportedLocations.push_back(newLocation);
+        listIndex = static_cast<uint32_t>(mSupportedMaps.size()) - 1; // new element is last in list
+        return true;
+    }
+
+    ChipLogError(Zcl, "AddSupportedLocation %u - supported locations list is already at maximum size %u", newLocation.locationID,
+                 static_cast<uint32_t>(kMaxNumSupportedLocations));
+
+    return false;
+}
+
+bool RvcServiceAreaDelegate::ModifySupportedLocation(uint32_t listIndex, const LocationStructureWrapper & modifiedLocation)
+{
+    // The server instance (caller) is responsible for ensuring that there are no duplicate location IDs, list size not exceeded,
+    // etc.
+
+    // Double-check that locationID's match.
+    if (modifiedLocation.locationID != mSupportedLocations[listIndex].locationID)
+    {
+        ChipLogError(Zcl, "ModifySupportedLocation - locationID's do not match, new locationID %u, existing locationID %u",
+                     modifiedLocation.locationID, mSupportedLocations[listIndex].locationID);
+        return false;
+    }
+
+    // checks passed, update the attribute
+    mSupportedLocations[listIndex] = modifiedLocation;
+    return true;
+}
+
+bool RvcServiceAreaDelegate::ClearSupportedLocations()
+{
+    if (!mSupportedLocations.empty())
+    {
+        mSupportedLocations.clear();
+        return true;
+    }
+
+    return false;
+}
+
+//*************************************************************************
+// Supported Maps accessors
+
+bool RvcServiceAreaDelegate::IsSupportedMapChangeAllowed()
+{
+    // TODO IMPLEMENT
+    return true;
+}
+
+uint32_t RvcServiceAreaDelegate::GetNumberOfSupportedMaps()
+{
+    return static_cast<uint32_t>(mSupportedMaps.size());
+}
+
+bool RvcServiceAreaDelegate::GetSupportedMapByIndex(uint32_t listIndex, MapStructureWrapper & aSupportedMap)
+{
+    if (listIndex < static_cast<uint32_t>(mSupportedMaps.size()))
```

Comment:
Why is cast needed?

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/87646aaae78294fe999f0491b3e5a91ec58adb9c).